demos/CMakeLists.txt

```diff
@@ -55,6 +55,14 @@ function(check_presigned_urls demo_name)
         set(CMAKE_REQUIRED_DEFINITIONS -DS3_PRESIGNED_GET_URL)
     endif()
 
+    if(S3_PRESIGNED_GET_URL)
+        target_compile_definitions(
+            ${demo_name} PRIVATE
+                S3_PRESIGNED_PUT_URL="${S3_PRESIGNED_PUT_URL}"
+        )
+        set(CMAKE_REQUIRED_DEFINITIONS -DS3_PRESIGNED_PUT_URL)
+    endif()
```

Comment:
Should be able to put this in the last if block

Reply:
Or perhaps you meant to set this as `if(S3_PRESIGNED_PUT_URL)`. In that case, I think the `if(S3_PRESIGNED_GET_URL AND S3_PRESIGNED_PUT_URL)` can be removed to make it possible for the user to define one parameter in the header file and another through a CMake flag.

Reply:
Yes thank you for the catch

Reply:
Looks like I still need the first if case:
![image](https://user-images.githubusercontent.com/6563840/97834858-af2f2100-1c8d-11eb-8da1-8833c1169546.png)